power_grid_model_c/power_grid_model/include/power_grid_model/main_core/update.hpp

```diff
@@ -82,6 +84,33 @@ inline void update_y_bus(YBus<sym>& y_bus, std::shared_ptr<MathModelParam<sym> c
     y_bus.update_admittance(math_model_param);
 }
 
+template <bool sym>
+inline void update_y_bus_increment(YBus<sym>& y_bus, std::shared_ptr<MathModelParam<sym> const> const& math_model_param,
+                                   bool increment) {
+    auto branch_param_to_change_views =
+        std::views::iota(0, math_model_param->branch_param.size()) | std::views::filter([&math_model_param](Idx i) {
```

Comment:
i did some research and our `libc++-11` STL implementation used by our linux compilers does not implement `std::views::iota` correctly yet (or rather, there was a [bug in the standard](https://cplusplus.github.io/LWG/issue3523) that they blindly (and therefore erroneously) copied. See the defect reports in the [CPP reference on `std::views::iota`](https://en.cppreference.com/w/cpp/ranges/iota_view). The bug in the standard was fixed, but not yet in `libc++-11`).

We'll have to wait until Ubuntu 24.04 comes out in April until we can use `std::views::iota`.

In the meantime, you can use `boost::counting_range` but it does not satisfy the requirements imposed by `std::views::view_interface` and therefore, `std::views::filter` can't be used 😢

Reply:
🤨ouch